### DISCUSSION TITLE
Nix Flake updates fails because of invalid GitHub credentials

### DISCUSSION THREAD

**Discussion by adisbladis** [renovatebot/renovate#23604] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm attempting to set up Renovate for several Nix based projects (https://github.com/nix-community/poetry2nix & https://github.com/adisbladis/pyproject.nix) and they are all failing to update with invalid GitHub credentials.

While this is not my project here is one which contains a publicly accessible error message: https://github.com/nix-community/lanzaboote/pull/216#issuecomment-1647075651.

As this problems occur on several independent repositories for multiple orgs/individuals I don't believe a minimal reproduction is necessary.

### Relevant debug logs

<details><summary>Logs</summary>

```

Command failed: docker run --rm --name=renovate_a_sidecar --label=renovate_a_child --memory=3584m -v "/tmp/worker/7cc045/2ecd68/repos/github/nix-community/lanzaboote":"/tmp/worker/7cc045/2ecd68/repos/github/nix-community/lanzaboote" -v "/tmp/worker/7cc045/2ecd68/cache":"/tmp/worker/7cc045/2ecd68/cache" -e CONTAINERBASE_CACHE_DIR -w "/tmp/worker/7cc045/2ecd68/repos/github/nix-community/lanzaboote" ghcr.io/containerbase/sidecar:9.6.1 bash -l -c "install-tool nix 2.16.1 && nix     --extra-experimental-features nix-command     --extra-experimental-features flakes --extra-access-tokens github.com=**redacted** flake update"
/usr/local/bin/docker: line 4: .: filename argument required
.: usage: . filename [arguments]
warning: '/nix/var/nix' does not exist, so Nix will use '/home/ubuntu/.local/share/nix/root' as a chroot store

�[K
�[Kwarning: ignoring untrusted flake configuration setting 'extra-substituters'.
Pass '--accept-flake-config' to trust it
warning: ignoring untrusted flake configuration setting 'extra-trusted-public-keys'.
Pass '--accept-flake-config' to trust it
error:
       … while updating the lock file of flake 'git+file:///tmp/worker/7cc045/2ecd68/repos/github/nix-community/lanzaboote?ref=refs/heads/master&rev=e7246c6bce1733d373059e6342f67fd53f90c198'

       … while updating the flake input 'flake-compat'

       … while fetching the input 'github:edolstra/flake-compat'

       error: unable to download 'https://api.github.com/repos/edolstra/flake-compat/commits/HEAD': HTTP error 401

       response body:

       {
         "message": "Bad credentials",
         "documentation_url": "https://docs.github.com/rest"
       }

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by maxime-cohere** [comment]
Duplicate/related to https://github.com/renovatebot/renovate/discussions/21567#discussioncomment-6085407

**Comment by adisbladis** [comment]
This has since been fixed and updates are working.
